clevertap-core/src/main/java/com/clevertap/android/sdk/cryption/AESGCMCrypt.kt

```diff
@@ -0,0 +1,127 @@
+package com.clevertap.android.sdk.cryption
+
+import android.os.Build
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import android.util.Base64
+import com.clevertap.android.sdk.Logger
+import java.nio.charset.StandardCharsets
+import java.security.KeyStore
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.GCMParameterSpec
+
+class AESGCMCrypt : Crypt() {
+
+    override fun encryptInternal(plainText: String): String? {
+        return performCryptOperation(
+            mode = Cipher.ENCRYPT_MODE,
+            data = plainText.toByteArray(StandardCharsets.UTF_8)
+        )?.let { (iv, encryptedBytes) ->
+            // Concatenate IV and encrypted text and surround with <>
+            "<${iv.toBase64()}${encryptedBytes.toBase64()}>"
+        }
+    }
+
+    override fun decryptInternal(cipherText: String): String? {
+        return parseCipherText(cipherText)?.let { (iv, encryptedBytes) ->
+            performCryptOperation(
+                mode = Cipher.DECRYPT_MODE,
+                data = encryptedBytes,
+                iv = iv
+            )
+        }?.let { (_, decryptedBytes) ->
+            String(decryptedBytes, StandardCharsets.UTF_8)
+        }
+    }
+
+    private fun parseCipherText(cipherText: String): Pair<ByteArray, ByteArray>? {
+        return try {
+            // removes the brackets <>
+            val content = cipherText.substring(1, cipherText.length - 1)
+            val ivLength = 16  // Base64 length for 12-byte IV (encoded length is 16)
+            val iv = content.substring(0, ivLength).fromBase64()
+            val encryptedBytes = content.substring(ivLength).fromBase64()
+            Pair(iv, encryptedBytes)
+        } catch (e: Exception) {
+            Logger.v("Error parsing cipherText", e)
+            null
+        }
+    }
+
+    private fun performCryptOperation(
+        mode: Int,
+        data: ByteArray,
+        iv: ByteArray? = null
+    ): Pair<ByteArray, ByteArray>? {
```

Comment:
Maybe having a data class like CryptResult(iv,data) would be easier to read instead of Pair<ByteArray,ByteArray>

Reply:
Addresses in [bab4cbb](https://github.com/CleverTap/clevertap-android-sdk/pull/719/commits/bab4cbbdf777894e45e9521a0c53b5e005d7c6c0)